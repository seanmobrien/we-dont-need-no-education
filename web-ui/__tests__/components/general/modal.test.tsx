import React from 'react';
import { render } from '/__tests__/test-utils';
import Modal from '/components/general/modal';

// Mock portal rendering
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (children: React.ReactNode) => children,
}));

describe('Modal', () => {
  const defaultProps = {
    isOpen: false,
    title: 'Test Modal Title',
    onClose: jest.fn(),
  };

  it('renders closed modal snapshot', () => {
    const { container } = render(
      <Modal {...defaultProps}>
        <div>Modal content</div>
      </Modal>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders open modal snapshot', () => {
    const { container } = render(
      <Modal {...defaultProps} isOpen={true}>
        <div>Modal content</div>
      </Modal>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders modal with title snapshot', () => {
    const { container } = render(
      <Modal {...defaultProps} isOpen={true} title="Test Modal Title">
        <div>Modal content</div>
      </Modal>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders modal with actions snapshot', () => {
    const actions = (
      <div>
        <button>Cancel</button>
        <button>Save</button>
      </div>
    );

    const { container } = render(
      <Modal {...defaultProps}>
        <div>Modal content</div>
      </Modal>,
    );
    expect(container).toMatchSnapshot();
  });
});
