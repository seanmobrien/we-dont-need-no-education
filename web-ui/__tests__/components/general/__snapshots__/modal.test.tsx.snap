// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Modal renders closed modal snapshot 1`] = `<div />`;

exports[`Modal renders modal with actions snapshot 1`] = `<div />`;

exports[`Modal renders modal with title snapshot 1`] = `
<div>
  <div
    class="css-1ei9n0w"
  >
    <div
      class="css-ebbbus"
    >
      <div
        class="css-zoykhr"
      >
        <h2
          class="css-1rbf9s1"
        >
          Test Modal Title
        </h2>
        <button
          class="css-1uizrlc"
        >
          ×
        </button>
      </div>
      <div
        class="css-a7r8ae"
      >
        <div>
          Modal content
        </div>
      </div>
      <div
        class="css-fvk0vs"
      >
        <button
          class="css-42vy0d"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders open modal snapshot 1`] = `
<div>
  <div
    class="css-1ei9n0w"
  >
    <div
      class="css-ebbbus"
    >
      <div
        class="css-zoykhr"
      >
        <h2
          class="css-1rbf9s1"
        >
          Test Modal Title
        </h2>
        <button
          class="css-1uizrlc"
        >
          ×
        </button>
      </div>
      <div
        class="css-a7r8ae"
      >
        <div>
          Modal content
        </div>
      </div>
      <div
        class="css-fvk0vs"
      >
        <button
          class="css-42vy0d"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
